### DISCUSSION TITLE
Hello, I'm trying to implement a plugin for opening a door

### DISCUSSION THREAD

**Discussion by barashF** [SmartlyDressedGames/Unturned-3.x-Community#3839] [Q&A]
Faced the problem that after applying door.updateToggle(open); the door opens, but the texture remains in place. On the advice, I tried to implement it through BarricadeManager.ServerSetDoorOpen, but the library does not have this method. Please tell me what I'm doing wrong, thanks in advance!

**Comment by TH3AL3X** [comment]
Show code...

**Comment by TH3AL3X** [comment]
Just check this from uEssentials
https://github.com/TH3AL3X/uEssentials/blob/265a67c35bab80a95b90e21dede132a5581f289a/src/Commands/CommandOpenDoor.cs#L35

Are you sure you imported the SDG and UnityEngine library?

```
InteractableDoorHinge hinge = hit.transform.GetComponent<InteractableDoorHinge>();
if (hinge != null)
{
    InteractableDoor door = hinge.door;
    bool open = !door.isOpen;

    BarricadeManager.ServerSetDoorOpen(door, open);
```

**Reply by barashF** [reply]
Thank you very much for your help, I was using an old version of SDG
